tests/shouldwork/Numbers/Bits.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE ViewPatterns #-}
+#if __GLASGOW_HASKELL__ < 900
+{-# LANGUAGE AllowAmbiguousTypes #-}
+#endif
+
+{-# OPTIONS_GHC -Wno-partial-type-signatures #-}
+
+{-
+This checks that the methods of the Bits and FiniteBits typeclasses
+do the same thing in HDL and Haskell
+-}
+module Bits where
+import Clash.Prelude
+import Clash.Explicit.Testbench
+import Data.Int
+import Data.Word
+import Data.Bits
+
+testBits :: forall a. (Num a, Bits a) => Integer -> Integer -> _
+testBits x0@(fromInteger -> x) y0 =
+  ( testBitsNoPopCount @a x0 y0
+  , popCount @a x
+  , bitSize @a x  -- deprecated and undefined for Integer
+  )
+
+testBitsNoPopCount :: forall a. (Num a, Bits a) => Integer -> Integer -> _
+testBitsNoPopCount (fromInteger @a -> x) y0@(fromInteger @a -> y) =
+  (
+    ( (.&.) @a x y
+    , (.|.) @a x y
+    , xor @a x y
+    , complement @a x
+    , shift @a x yInt
+    , rotate @a x yInt
+    , zeroBits @a
+    )
+  , ( bit @a yInt
+    , setBit @a x yInt
+    , clearBit @a x yInt
+    , complementBit @a x yInt
+    , testBit @a x yInt
+    , bitSizeMaybe @a x
+    , isSigned @a x
+    )
+  , ( shiftL @a x yInt
+    , unsafeShiftL @a x yInt
+    , shiftR @a x yInt
+    , unsafeShiftR @a x yInt
+    , rotateL @a x yInt
+    , rotateR @a x yInt
+    )
+  )
+ where
+  yInt :: Int
+  yInt = (abs (fromInteger y0)) `mod` sz
+  sz = case bitSizeMaybe x of
+    Nothing -> 64
+    Just x -> fromIntegral x
+
+testFiniteBits :: forall a. (Num a, FiniteBits a) => Integer -> Integer -> _
+testFiniteBits x0@(fromInteger -> x) y0@(fromInteger -> y) =
+  ( testBits @a x0 y0
+  , finiteBitSize @a x
+  , countLeadingZeros @a x
+  , countTrailingZeros @a x
+  )
+
+topEntity1 :: (Integer, Integer) -> _
+topEntity1 (x,y) =
+    ( testBitsNoPopCount @Integer x y
+    , testFiniteBits @Int x y
+    , testFiniteBits @Int8 x y
+    , testFiniteBits @Int16 x y
+    -- , testFiniteBits @Int32 x y -- TODO triggers GHDL bug #2618, re-enable after we get another GHDL
```

Comment:
Could you also add a comment here https://github.com/clash-lang/clash-compiler/blob/b9212126339fe3a587c4f4d701bb7a87e73ef3a7/.ci/docker/Dockerfile#L24 ?